extensions/kotlin/deployment/src/main/java/io/quarkus/kotlin/deployment/KotlinProcessor.java

```diff
@@ -30,12 +50,41 @@ void registerKotlinJacksonModule(BuildProducer<ClassPathJacksonModuleBuildItem>
         }
     }
 
-    /**
-     * Kotlin data classes that have multiple constructors need to have their final fields writable,
-     * otherwise creating a instance of them with default values, fails in native mode
+    /*
+     * Register the Kotlin reflection types if they are present.
      */
     @BuildStep
-    ReflectiveClassFinalFieldsWritablePredicateBuildItem dataClassPredicate() {
-        return new ReflectiveClassFinalFieldsWritablePredicateBuildItem(new IsDataClassWithDefaultValuesPredicate());
+    void registerKotlinReflection(final BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
```

Comment:
Why go to all this trouble when you can just create build items with the values present in the files?

There is no benefit to reading the file from the classpath  and loading Jackson to do so... It's actually quite slow...

Reply:
yeah, i could I suppose.  I was thinking drop in replacements of those files as they evolve.  But i suppose either way is editing a text file of one sort of another.  Speed, however, not much of an issue here.  ;)